api/src/middleware/cache.ts

```diff
@@ -0,0 +1,18 @@
+import { Request, Response, NextFunction } from "express"
+
+/**
+ * Cache-Control header tells the browser to cache it for a week,
+ * and Vercel's Edge CDN caches it via https://vercel.com/docs/edge-network/caching
+ */
+export function withCacheHeader(
+  _req: Request,
+  res: Response,
+  next: NextFunction,
+) {
+  res.set(
+    "Cache-Control",
+    "public, max-age=604800, s-maxage=604800, stale-while-revalidate=18000",
```

Comment:
`s-maxage` and `stale-while-revalidate` is for Vercel, while `max-age` is for the Browser